src/test/ui/feature-gate/issue-43106-gating-of-builtin-attrs-error.stderr

```diff
@@ -53,84 +53,151 @@ LL | |
 LL | | }
    | |_- not a function or static
 
+error: attribute should be applied to a function or static
+  --> $DIR/issue-43106-gating-of-builtin-attrs-error.rs:23:1
+   |
+LL | / #![deny(unused_attributes)]
+LL | |
+LL | |
+LL | |
+...  |
+LL | | #![no_mangle]
+   | | ^^^^^^^^^^^^^
+...  |
+LL | |
+LL | | fn main() {}
+   | |____________- not a function or static
```

Comment:
Highlighting the entire crate span is a bit confusing here. I've not tested but maybe passing `DUMMY_SP` to `check_attributes` instead of `tcx.hir().span(CRATE_HIR_ID)` might work to clean this up.